schema/me/home_page_modules.js

```diff
@@ -0,0 +1,151 @@
+  GraphQLList,
+  GraphQLObjectType,
+  GraphQLString,
+} from 'graphql';
+
+const RESULTS_SIZE = 15;
+
+const featuredFair = () => {
+  return gravity('fairs', { size: 5, status: 'running' }).then((fairs) => {
+    if (fairs) {
+      return first(sortBy(fairs, ({ banner_size }) =>
+        ['x-large', 'large', 'medium', 'small', 'x-small'].indexOf(banner_size)
+      ));
+    }
+  });
+};
+
+const featuredAuction = () => {
+  return gravity('sales', { live: true, size: 1 }).then((sales) => {
+    if (sales) {
+      return first(sales);
+    }
+  });
+};
+
+const moduleTitle = {
+  active_bids: () => 'Your Active Bids',
+  followed_artists: () => 'Works by Artists you Follow',
+  followed_galleries: () => 'Works from Galleries you Follow',
+  saved_works: () => 'Recently Saved Works',
+  recommended_works: () => 'Recommended Works for You',
+  live_auctions: () => {
+    return featuredAuction().then(({ name }) => {
+      return `At auction: ${name}`;
+    });
+  },
+  current_fairs: () => {
+    return featuredFair().then(({ name }) => {
+      return `Art Fair: ${name}`;
+    });
+  },
+  related_artists: () => 'Works by Related Artists',
+  genes: () => 'Works from Gene you Follow',
+};
+
+const moduleResults = {
+  active_bids: () => [],
+  followed_artists: (accessToken) => {
+    return gravity
+      .with(accessToken)('me/follow/artists/artworks', {
+        for_sale: true,
+        size: RESULTS_SIZE,
+      });
+  },
+  followed_galleries: (accessToken) => {
+    return gravity.with(accessToken)('me/follow/profiles/artworks', {
+      for_sale: true,
+      size: RESULTS_SIZE,
+    });
+  },
+  saved_works: (accessToken) => {
+    return gravity.with(accessToken)('me').then((user) => {
+      return gravity
+        .with(accessToken)('collection/saved-artwork/artworks', {
+          size: RESULTS_SIZE,
+          user_id: user.id,
+        });
+    }).catch(() => {});
+  },
+  recommended_works: (accessToken) => {
+    return gravity.with(accessToken)('me/suggested/artworks/homepage', { limit: RESULTS_SIZE });
+  },
+  live_auctions: () => {
+    return featuredAuction().then(({ id }) => {
+      return gravity(`sale/${id}/sale_artworks`, { size: RESULTS_SIZE });
+    });
+  },
+  current_fairs: () => {
+    return featuredFair().then(({ id }) => {
+      return gravity('filter/artworks', {
+        fair_id: id,
+        for_sale: true,
+        size: RESULTS_SIZE,
+      }).then(({ hits }) => hits);
+    });
+  },
+  related_artists: () => [],
+  genes: () => [],
+};
+
+
+export const HomePageModulesType = new GraphQLObjectType({
+  name: 'HomePageModules',
+  fields: () => ({
+    key: {
+      type: GraphQLString,
+    },
+    display: {
+      type: GraphQLString,
+    },
+    title: {
+      type: GraphQLString,
+      resolve: ({ key, display }, options, { rootValue: { accessToken } }) => {
+        if (display) return moduleTitle[key](accessToken);
+      },
+    },
+    results: {
+      type: new GraphQLList(Artwork.type),
```

Comment:
There's nothing preventing you from doing so in that case, use a `GraphQLUnionType`